### PR TITLE
Add task for Kube-Linter Tool

### PR DESCRIPTION
**Pull request by urvashigupta7**
This task makes it possible to use Kube-Linter within Tekton pipelines.
# Submitter Checklist
- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

### CODE REVIEW

**Comment by urvashigupta7** [comment]
@vinamra28 Can you please enable tests over here?

File: task/kube-linter/0.1/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+## Workspaces
+* **source** : A Workspace containing your source directory.
+
+## Parameters
+* **config_file_url** : url from where the config file would be fetched.
+* **config_file_path** : path to config file(written in YAML) for custom checks.
+* **manifest** : path to manifest files or manifest directory to be checked. (default: `.`)
+* **includelist** : checks to be included.
+* **excludelist** : checks to be excluded.
+* **default-option** : provides two options (adding all built-in checks or disabling all default checks): add-all-built-in and/do-not-auto-add-defaults.
```
Comment (GijsvanDulmen) [inline task/kube-linter/0.1/README.md:25]: Typo? Should be `default_option`?

Comment (urvashigupta7) [reply task/kube-linter/0.1/README.md]: Corrected the typo.

File: task/kube-linter/0.1/kube-linter.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: kube-linter
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.19.0"
```
[the remark this answers is not in the document]
Comment (urvashigupta7) [reply task/kube-linter/0.1/kube-linter.yaml]: I don't think so..Updated the version

File: task/kube-linter/0.1/kube-linter.yaml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      type: array
+      description: "args"
+      default: []
+  steps:
+    - name: fetch-config-file
+      image: registry.access.redhat.com/ubi8/ubi-minimal:8.2
+      workingDir: $(workspaces.source.path)
+      script: |
+         #!/usr/bin/env bash
+
```
Comment (chmouel) [inline task/kube-linter/0.1/kube-linter.yaml:60]: maybeadd a `set -e` here to make this step more robust?

File: task/kube-linter/0.1/kube-linter.yaml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+         else
+          echo "No config file url was set"
+         fi
+
+    - name: lint-yaml
+      image: docker.io/stackrox/kube-linter:0.2.2-2-g7d10a69154-alpine@sha256:e520e9d8d3a2dfa611914836536545b135845e7bda9f1df34b060e116232dbf0
+      workingDir: $(workspaces.source.path)
+      script: |
+        #!/bin/sh
+
```
Comment (chmouel) [inline task/kube-linter/0.1/kube-linter.yaml:74]: You don't have to add the "shbang" line `#!/bin/sh` tekton would automatically add it and add some "set -eux"  at the same time,

File: task/kube-linter/0.1/samples/pvc.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1000Mi
```
Comment (chmouel) [inline task/kube-linter/0.1/samples/pvc.yaml:10]: is 1GB needed for example?

**Review by chmouel — COMMENTED** [review]
some nits, 

thanks for your contribution,

**Comment by chmouel** [comment]
/lgtm

🚀

**Review by GijsvanDulmen — APPROVED** [review]

**Pull request merged**